File: DS and Algorithm/AlgoExpert/Easy/branchSums.js
function branchSums(roor) {
    let sum = [];
    calculateBranchSums(root, 0, sum);
    return sum;
}

function calculateBranchSums(node, runningSum, sums) {
    if (node === null) {
        return
    }
    newRunningSum = runningSum + node.value;
    if (node.left === null && node.right === null) {
        sums.push(newRunningSum)
    }
    calculateBranchSums(node.left, newRunningSum, sums);
    calculateBranchSums(node.right, newRunningSum, sums)
}

File: DS and Algorithm/AlgoExpert/Easy/branchSums.py
def branchSUms(root):
    sums = []
    calculateBranchSums(root, 0, sums)
    return sums


def calculateBranchSums(node, runningSum, sums):
    if node is None:
        return
    nodeRunningSum = runningSum + node.value
    if node.left is None and node.right is None:
        sums.append(nodeRunningSum)
        return

    calculateBranchSums(node.left, nodeRunningSum, sums)
    calculateBranchSums(node.runningSum, nodeRunningSum, sums)
